src/aave-v2/lens/RatesLens.sol

```diff
@@ -313,6 +336,45 @@ abstract contract RatesLens is UsersLens {
     /// @return poolSupplyRate The market's pool supply rate per year (in ray).
     /// @return poolBorrowRate The market's pool borrow rate per year (in ray).
     function getRatesPerYear(address _poolToken)
+        public
+        view
+        returns (
+            uint256,
+            uint256,
+            uint256,
+            uint256
+        )
+    {
+        return _getRatesPerYear(_poolToken, 0, 0, 0, 0);
+    }
+
+    /// INTERNAL ///
+
+    struct PoolRatesVars {
+        uint256 availableLiquidity;
+        uint256 totalStableDebt;
+        uint256 totalVariableDebt;
+        uint256 avgStableRate;
+        uint256 reserveFactor;
+    }
+
+    /// @notice Computes and returns peer-to-peer and pool rates for a specific market.
+    /// @param _poolToken The market address.
+    /// @param _suppliedOnPool The amount hypothetically supplied to the underlying's pool.
+    /// @param _borrowedFromPool The amount hypothetically borrowed from the underlying's pool.
+    /// @param _repaidOnPool The amount hypothetically repaid to the underlying's pool.
+    /// @param _withdrawnFromPool The amount hypothetically withdrawn from the underlying's pool.
+    /// @return p2pSupplyRate The market's peer-to-peer supply rate per year (in ray).
+    /// @return p2pBorrowRate The market's peer-to-peer borrow rate per year (in ray).
+    /// @return poolSupplyRate The market's pool supply rate per year (in ray).
+    /// @return poolBorrowRate The market's pool borrow rate per year (in ray).
+    function _getRatesPerYear(
+        address _poolToken,
+        uint256 _suppliedOnPool,
+        uint256 _borrowedFromPool,
+        uint256 _repaidOnPool,
+        uint256 _withdrawnFromPool
```

Comment:
We usually use the order 1. supply 2. borrow 3. withdraw, 4. repay, see for example the structs `MaxGasForMatching` & `MarketPauseStatus` in `Types.sol` or the order of the main functions in `Morpho.sol`